File: OMC-forExperts/OMC-186/OMC186-B.py
# import
import math

# 問題にある関数の実装
def f(N):
    return (math.ceil(math.sqrt(N)))**2 - ( math.floor(math.sqrt(N)))**2
        
# Answer
ans = 1
# calcurate result
num = 0
# Limit
N = 10**5

# calcuration
while True:
    num += f(ans)
    if num >= N:
        break
    ans += 1
    
# output
print("ans is:",ans)

File: OMC-forExperts/OMC-186/OMC186-B.cpp
#include <iostream>
#include <cmath>

// function f
int f(int N) {
    return (int)(std::ceil(std::sqrt(N))) * (int)(std::ceil(std::sqrt(N))) 
           - (int)(std::floor(std::sqrt(N))) * (int)(std::floor(std::sqrt(N)));
}

int main() {
    // Answer
    int ans = 1;
    // Calculate result
    int num = 0;
    // Limit
    int N = 100000;

    // Calculation
    while (true) {
        num += f(ans);
        if (num >= N) {
            break;
        }
        ans++;
    }

    // Output
    std::cout << "ans is: " << ans << std::endl;

    return 0;
}